script/benchmark.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# https://ruby-doc.org/stdlib-2.5.0/libdoc/benchmark/rdoc/Benchmark.html
+# https://github.com/evanphx/benchmark-ips
+# https://github.com/SamSaffron/memory_profiler
+# https://github.com/tmm1/stackprof
+
+require_relative '../config/environment'
+
+def benchmark(name: nil, time_threshold: 3600, setup: nil, teardown: nil, &block)
+  ActionMailer::Base.perform_deliveries = false
+
+  timestamp = Time.now.utc.iso8601
+  stackprof_file = "script/benchmarks/output/#{name}_#{timestamp}_CPU.dump"
+  flamegraph_file = "script/benchmarks/output/#{name}_#{timestamp}_FLM"
+  memprof_file = "script/benchmarks/output/#{name}_#{timestamp}_MEM.log"
+  benchmark_file = "script/benchmarks/output/#{name}_#{timestamp}_BCM.log"
+  $stdout = File.new(benchmark_file, 'w') if ENV['CAP_STDOUT']
+  $stdout.sync = true if ENV['CAP_STDOUT']
+
+  benchmark_report = nil
+
+  ActiveRecord::Base.transaction do
```

Comment:
Future versions of Rails will have some performance optimizations based on if queries are run in transactions. Is there any way that we can remove the transaction and get the database back into its original state between benchmarks instead of a transaction?

Reply:
See latest changes that address this